### PR TITLE
[DS-2648] Full text available sidebar facet for Discovery

### PR DESCRIPTION
**Pull request by christian-scheible**
This PR adds a facet to xmlui and jspui which makes it possible to filter by full text available. 
In this version the 
- XMLUI with the Mirage theme is localized and it looks like: <br />
  image
- In Mirage 2 it should be localized too but wasn't tested
- In JSPUI it's not localized and always shows the labels true/false: <br />
  image

To make the changes of this pull request visible:
1. merge it 
2. run 
   <pre>
   [dspace]/bin/dspace index-discovery -b
   </pre>
3. clear the xmlui cache
4. check the DSpace home page

**Remark**
- If this pull request get merged I think the configuration in discovery.xml should not be the default but become part of the documentation (which I will do if this get's merged)
- To localize the JSPUI part I think the following line should be the starting point:
  https://github.com/DSpace/DSpace/blob/master/dspace-jspui/src/main/webapp/discovery/static-sidebar-facet.jsp#L102

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/discovery/SolrServiceFullTextFilterPlugin.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/**
```
Comment (christian-scheible) [inline dspace-api/src/main/java/org/dspace/discovery/SolrServiceFullTextFilterPlugin.java]: This adds the required fields to Solr

File: dspace/solr/search/conf/schema.xml [hunk 2]
```diff
@@ -537,6 +537,8 @@
       
    <field name="read" type="string" indexed="true" stored="true" omitNorms="true" multiValued="true" docValues="true" />
```
Comment (christian-scheible) [inline dspace/solr/search/conf/schema.xml]: Be sure to avoid stemming for the values: true and false

**Comment by tdonohue** [comment]
I like this feature/idea, but I don't think it actually is finding "full text available" items. All it's doing is finding items that have files attached. There's no real guarantee that the file attached *is* a Full Text document though. The attached file could be a PDF abstract, or an image, or a video, or a dataset. 

I think this is a good feature, but it needs to be renamed to be a "Has File(s)" facet... or "Has Content" or similar.

**Comment by tdonohue** [comment]
Hi @christian-scheible , 

According to Travis CI, there's a compilation error against "master".  Here's the error it's reporting:
```
[ERROR] /home/travis/build/DSpace/DSpace/dspace-api/src/main/java/org/dspace/discovery/SolrServiceContentInOriginalBundleFilterPlugin.java:[46,31] method getBundles in class org.dspace.content.Item cannot be applied to given types;
  required: no arguments
  found: java.lang.String
  reason: actual and formal argument lists differ in length
```
This  *might* have to do with the fact that we (very) recently refactored the `dspace-api` in preparation for 6.0.  See https://wiki.duraspace.org/display/DSPACE/DSpace+Service+based+api

So, you may want to be sure you are working against the latest version of `master`, as the DSpace API has been refactored significantly.  If you run into issues, or have questions, let us know!

Thanks for your hard work on this!

**Comment by hardyoyo** [comment]
I'll note that the pull request has been amended to address tdonohue's concerns, it's ready for review.

File: dspace/solr/search/conf/schema.xml (hunk 2 above)
Comment (KevinVdV) [inline dspace/solr/search/conf/schema.xml]: Would a type boolean not work ? (Because it is only intended to store booleans)

**Comment by KevinVdV** [comment]
Added a small note here: https://github.com/DSpace/DSpace/pull/988/files#r47216109
I also noticed that in the JSPUI you get a "next >" url in the "Has file(s)" box, not sure what this represents, when I click on it the facet values are gone from the view.

If the JSPUI gets fixed, this is a +1 from me.

**Comment by helix84** [comment]
+1, I tested in XMLUI and JSPUI and it works nicely. If it's to be enabled out of the box, the values should be localized in JSPUI, too.

The "Next >" link that Kevin mentioned appears in JSPUI in the fulltext facet box after you click on another facet (e.g. subject).
https://github.com/christian-scheible/DSpace/blob/DS-2648/dspace-jspui/src/main/webapp/search/discovery.jsp#L740

**Comment by kohts** [comment]
This is extremely useful feature in my opinion. Tested both in xmlui and jspui.

Mentioned "next >" jspui issue seems to me more like an already existing jspui bug,
rather than something which this PR introduces (specific, of course, to the facet
parameters which this PR uses, specifically facetLimit and maybe solr field datatype).

Would be a pity to miss this feature just because of cosmetic issue in the UI.

**Comment by KevinVdV** [comment]
The bug that I found is **not related to this pull request**, it is a general JSPUI bug that is reported here: https://jira.duraspace.org/browse/DS-2959

This pull requests gets a +1 from me, but I wonder if we should enable the feature by default ? The author ( @christian-scheible mentioned this as well ). Depending on the outcome we could await the disable by default or merge directly.

**Comment by helix84** [comment]
I'm +1 on enabled by default.

**Comment by hardyoyo** [comment]
I'm also +1 enabled by default

**Pull request merged**